src/devices/sound/mas3507d.h

```diff
@@ -25,7 +25,23 @@ class mas3507d_device : public device_t, public device_sound_interface
 	void reset_sample_count() { total_sample_count = 0; }
 	u32 get_sample_count() const { return total_sample_count; }
 
-	void set_playback_enabled(bool enabled) { playback_enabled = enabled; }
+	void set_playback_enabled(bool enabled) {
+		playback_enabled = enabled;
+		reset_sample_count();
+
+		memset(mp3data.data(), 0, mp3data.size());
```

Comment:
The mp3data buffer needs to be cleared and mp3_count reset or else the previously played audio file will play until the buffers are filled with the new MP3's data.